src/HotChocolate/Core/test/Types.Tests/Configuration/TypeDiscoveryTests.cs

```diff
@@ -37,6 +37,15 @@ public void TypeDiscovery_Should_InferStructs()
             .MatchSnapshot();
     }
 
+    [Fact]
+    public void InferInputStructsWithNonDefaultCtor()
+    {
+        SchemaBuilder.New()
```

Comment:
This test is not sufficient. What are we testing here? While a schema snapshot test can be a good start this one essentially is fire and forget.

Reply:
I thought this would be sufficient because schema build throws an exception without the fix, but I've added the schema snapshot.